packages/client/src/api/index.js

```diff
@@ -30,10 +35,12 @@ export const createApi = ({ rootPath = "", setState, getState }) => {
     }
   }
 
-  const post = apiCall("POST")
+  const dummyCall = () => ({})
+
+  const post = isBuilderPreview ? dummyCall : apiCall("POST")
   const get = apiCall("GET")
-  const patch = apiCall("PATCH")
-  const del = apiCall("DELETE")
+  const patch = isBuilderPreview ? dummyCall : apiCall("PATCH")
```

Comment:
What we are intending to do in the builder preview is to disable click events. We can remove all the `isBuilderPreview` stuff here, and simply disable click events inside the iframe, which should give us the same result:

```js
iframe.onclick = function(e) {
                e.preventDefault();
                return false;
}
```

should be all the code you need.